libflux/src/semantic/nodes.rs

```diff
@@ -0,0 +1,567 @@
+
+    Int(IntegerLiteral<'a>),
+    Flt(FloatLiteral<'a>),
+    Str(StringLiteral<'a>),
+    Dur(DurationLiteral<'a>),
+    Uint(UnsignedIntegerLiteral<'a>),
+    Bool(BooleanLiteral<'a>),
+    Time(DateTimeLiteral<'a>),
+    Regexp(RegexpLiteral<'a>),
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum PropertyKey<'a> {
+    Identifier(Identifier<'a>),
+    StringLiteral(StringLiteral<'a>),
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum FunctionBody<'a> {
+    Block(Block<'a>),
+    Expr(Expression<'a>),
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct Package<'a> {
+    pub loc: Location<'a>,
+
+    pub package: String,
+    pub files: Vec<File<'a>>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct File<'a> {
+    pub loc: Location<'a>,
+
+    pub package: Option<PackageClause<'a>>,
+    pub imports: Vec<ImportDeclaration<'a>>,
+    pub body: Vec<Statement<'a>>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct PackageClause<'a> {
+    pub loc: Location<'a>,
+
+    pub name: Identifier<'a>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct ImportDeclaration<'a> {
+    pub loc: Location<'a>,
+
+    pub alias: Option<Identifier<'a>>,
+    pub path: StringLiteral<'a>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct Block<'a> {
+    pub loc: Location<'a>,
+
+    pub body: Vec<Statement<'a>>,
+}
+
+impl Block<'_> {
+    fn return_statement(&self) -> &ReturnStatement {
+        let len = self.body.len();
+        let last = self
+            .body
+            .get(len - 1)
+            .expect("body must have at least one statement");
+        let last: Option<&ReturnStatement> = match last {
+            Statement::Ret(rs) => Some(rs),
+            _ => None,
+        };
+        last.expect("last statement must be a return statement")
+    }
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct OptionStatement<'a> {
+    pub loc: Location<'a>,
+
+    pub assignment: Assignment<'a>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct BuiltinStatement<'a> {
+    pub loc: Location<'a>,
+
+    pub id: Identifier<'a>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct TestStatement<'a> {
+    pub loc: Location<'a>,
+
+    pub assignment: VariableAssignment<'a>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct ExpressionStatement<'a> {
+    pub loc: Location<'a>,
+
+    pub expression: Expression<'a>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct ReturnStatement<'a> {
+    pub loc: Location<'a>,
+
+    pub argument: Expression<'a>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct VariableAssignment<'a> {
+    pub loc: Location<'a>,
+
+    pub id: Identifier<'a>,
+    pub init: Expression<'a>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct MemberAssignment<'a> {
+    pub loc: Location<'a>,
+
+    pub member: MemberExpression<'a>,
+    pub init: Expression<'a>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct StringExpression<'a> {
+    pub loc: Location<'a>,
+    pub type_info: TypeInformation,
```

Comment:
Could this field just be of type `MonoType`?

Reply:
see https://github.com/influxdata/flux/pull/2027#discussion_r336613955